File: AnotherYetBashClient/src/main/java/ru/aim/anotheryetbashclient/BashApp.java
package ru.aim.anotheryetbashclient;

import android.app.Application;

import ru.aim.anotheryetbashclient.helper.DbHelper;

public class BashApp extends Application {

    private static BashApp instance;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        new Thread(new Runnable() {
            @Override
            public void run() {
                DbHelper.getInstance(BashApp.this);
            }
        }).start();
    }

    public static BashApp getInstance() {
        return instance;
    }
}

File: AnotherYetBashClient/src/main/java/ru/aim/anotheryetbashclient/BashApp.kt
package ru.aim.anotheryetbashclient

import android.app.Application

import ru.aim.anotheryetbashclient.data.DaggerDataComponent
import ru.aim.anotheryetbashclient.data.DataComponent
import ru.aim.anotheryetbashclient.data.source.local.LocalDataSourcesModule
import ru.aim.anotheryetbashclient.data.source.remote.RemoteDataSourceModule
import ru.aim.anotheryetbashclient.data.source.remote.parser.ParserModule

class BashApp : Application() {

    lateinit var dataComponent: DataComponent

    override fun onCreate() {
        super.onCreate()

        val appModule = ApplicationModule(this)
        val local = LocalDataSourcesModule()
        val remote = RemoteDataSourceModule()
        val parser = ParserModule()

        dataComponent = DaggerDataComponent
                .builder()
                .applicationModule(appModule)
                .localDataSourcesModule(local)
                .remoteDataSourceModule(remote)
                .parserModule(parser)
                .build()

        println(dataComponent)
    }
}